src/borg/cache.py

```diff
@@ -452,25 +455,34 @@ def chunk_decref(self, id, stats):
         else:
             stats.update(-size, -csize, False)
 
-    def file_known_and_unchanged(self, path_hash, st, ignore_inode=False):
+    def file_cache_key(self, hash, path, st, ignore_inode):
+        if ignore_inode or fstype(path) not in FS_WITH_STABLE_INODES:
+            # we don't use the path directly (but its hash) to safe memory
+            cache_key = hash(safe_encode(path))
```

Comment:
So this would be incompatible with existing files caches (incompatible in the sense of rechunking / data is worthless). Since the keys are different it would also ~double the on-disk size until 10 (cache TTL) archives are created.